### PR TITLE
Code/test port to .NET Core RC2

### PR DESCRIPTION
**Pull request by nblumhardt**
Work still required on _Build.ps1_

### CODE REVIEW

File: src/Serilog/project.json [hunk 1]
```diff
@@ -1,57 +1,54 @@
...
+    "keyFile": "../../assets/Serilog.snk",
+    "xmlDoc": true
   },
   "frameworks": {
-    "net45": {
-      "compilationOptions": {
-        "define": [ "PROCESS", "REMOTING", "APPDOMAIN" ]
+    "net4.5": {
+      "buildOptions": {
+        "define": [ "REMOTING" ]
```
Comment (adamchester) [inline src/Serilog/project.json]: Did you intend to remove "PROCESS" and "APPDOMAIN" ?

File: test/Serilog.Tests/Support/StringSink.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+using Serilog.Formatting.Display;
+
+namespace Serilog.Tests.Support
+{
+    public class StringSink : ILogEventSink
+    {
+        readonly StringWriter _sw = new StringWriter();
+        readonly ITextFormatter _formatter;
+
+        const string DefaultOutputTemplate = "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level}] {Message}{NewLine}{Exception}";
```
Comment (adamchester) [inline test/Serilog.Tests/Support/StringSink.cs]: Should `StringSink` exclude the `{Timestamp}` to make testing easier?

File: src/Serilog/project.json (hunk 1 above)
Comment (nblumhardt) [inline src/Serilog/project.json]: Yes, the core Serilog package doesn't depend on either :+1:

File: test/Serilog.Tests/Support/StringSink.cs (hunk 2 above)
Comment (nblumhardt) [inline test/Serilog.Tests/Support/StringSink.cs]: Probably depends on the test, seems like a good change but trying to keep the port 1:1 for now.

File: src/Serilog/project.json [hunk 3]
```diff
@@ -1,57 +1,54 @@
...
-    "dotnet5.4": {
-      "compilationOptions": {
-        "define": [ "ASYNCLOCAL", "PROCESS", "USERNAMEFROMENV" ]
-
+    "netstandard1.3": {
+      "buildOptions": {
+        "define": [ "ASYNCLOCAL" ]
       },
       "dependencies": {
-        "Microsoft.CSharp": "4.0.1-beta-23516",
```
Comment (merbla) [inline src/Serilog/project.json]: Bye bye crappy dependecies :)

**Comment by merbla** [comment]
All good, running on OSX...

```
$ dotnet build
Project Serilog (.NETFramework,Version=v4.5) will be compiled because expected outputs are missing
Compiling Serilog for .NETFramework,Version=v4.5

......

Compilation succeeded.
    26 Warning(s)
    0 Error(s)

Time elapsed 00:00:05.1382043


Project Serilog (.NETStandard,Version=v1.0) will be compiled because expected outputs are missing
Compiling Serilog for .NETStandard,Version=v1.0

......

Compilation succeeded.
    27 Warning(s)
    0 Error(s)

Time elapsed 00:00:04.3800085


Project Serilog (.NETStandard,Version=v1.3) will be compiled because expected outputs are missing
Compiling Serilog for .NETStandard,Version=v1.3 

......
Compilation succeeded.
    27 Warning(s)
    0 Error(s)

Time elapsed 00:00:04.7565901
```

:shipit:

**Comment by nblumhardt** [comment]
Ready to merge.

**Comment by SimonOrdo** [comment]
Issues using this in Framework Core project (standard Visual Studio RC2 Web project template). Fails with error:

    Package Serilog.Extensions.Logging 1.0.0-rc2-10096 is not compatible with net452 (.NETFramework,Version=v4.5.2). Package Serilog.Extensions.Logging 1.0.0-rc2-10096 supports: netstandard1.3 (.NETStandard,Version=v1.3)
    One or more packages are incompatible with .NETFramework,Version=v4.5.2.

**Comment by nblumhardt** [comment]
@SimonOrdo thanks for the heads-up. This is: https://github.com/serilog/serilog-extensions-logging/issues/31

**Comment by khellang** [comment]
:clap:

**Pull request merged**